packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesModule.kt

```diff
@@ -282,9 +283,15 @@ class UpdatesModule(
                   } else {
                     updatesServiceLocal.resetSelectionPolicy()
                     updateInfo.putBoolean("isNew", true)
+
+                    // We need the explicit casting here because when in versioned expo-updates,
+                    // the UpdateEntity and UpdatesModule are in different package namespace,
+                    // Kotlin cannot do the smart casting for that case.
+                    val updateEntity = loaderResult.updateEntity as UpdateEntity
```

Comment:
without this, kotlin will show a error that it cannot do the smart casting. since they are cross packages from versioned `abi49_0_0.expo.modules.updates.UpdatesModule` to unversioned `expo.modules.updates.db.entity.UpdateEntity`. we have to do the explicitly casting here.